src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1132,13 +1132,14 @@ internal virtual void IntersectWithEnumerable(IEnumerable<T> other)
                 if (Contains(item))
                 {
                     toSave.Add(item);
+                    DoRemove(item);
                 }
             }
 
-            if (toSave.Count < Count)
+            Clear();
+            foreach (T item in toSave)
             {
-                Clear();
-                AddAllElements(toSave);
+                AddIfNotPresent(item);
```

Comment:
> Changing AddAllElements would break the logic as it's used from UnionWith which needs to check for Contains before adding.

Can you help me understand this?  Doesn't `Add` only add if it the set doesn't already contain the item being added (i.e. it implicitly does the equivalent of Contains)?  If so, what is the functional difference between:
```C#
private void AddAllElements(IEnumerable<T> collection)
{
    foreach (T item in collection)
    {
        if (!Contains(item))
        {
            Add(item);
        }
}
```
and:
```C#
private void AddAllElements(IEnumerable<T> collection)
{
    foreach (T item in collection)
    {
        Add(item);
    }
}
```
?  I'm obviously missing something.